### PR TITLE
fix: ban function `now` in generated column 

### LINKED ISSUE #9571: bug: should not allow `NOW` in generated columns
In RisingWave, the `NOW` function under streaming execution behaves like an updatable stream of the latest timestamp, while the `PROCTIME` function means a snapshot of the timestamp when the record is processed.

Currently, we can create a source or table with a generated column of `NOW`, which has some semantics issues as above. We should not allow this from happening and encourage users to use `PROCTIME` here.

```
dev=> CREATE TABLE t3 (v INT, t timestamptz as now()) WITH (
    connector = 'datagen',
    fields.v.kind = 'sequence',
    fields.v.start = '1',
    fields.v.end  = '10',
    datagen.rows.per.second='15',
    datagen.split.num = '1'
) ROW FORMAT JSON;
ERROR:  QueryError: internal error: Rpc error: gRPC error (Internal error): Expression error: Unsupported function: Ok(Now)


dev=> CREATE SOURCE t3 (v INT, t timestamptz as now()) WITH (
    connector = 'datagen',
    fields.v.kind = 'sequence',
    fields.v.start = '1',
    fields.v.end  = '10',
    datagen.rows.per.second='15',
    datagen.split.num = '1'
) ROW FORMAT JSON;
CREATE_SOURCE
dev=> create materialized view mv as select * from t3;
ERROR:  QueryError: Expr error: Unsupported function: Ok(Now)
```

Note that we do report errors here but it's not that intuitive. Instead, the error messages reveal another issue of the frontend. Previously we handled these time-related expressions by rewriting them with constants during the binding phase. However, it's now possible that some expressions reside in the table's catalog, so we fail to catch them. This may also affect select-from-source of `PROCTIME`, or these functions in default column values.

cc @yuhao-su @Eurekaaw

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- The semantic `now` in generated column is very misleading. We encourage using proc_time()

close #9571

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: e2e_test/batch/basic/table_with_default_columns.slt.part [hunk 1]
```diff
@@ -44,7 +44,7 @@ select * from t2;
 2 2
 
 statement ok
-create table tt (v1 int, v2 timestamptz default now());
+create table tt (v1 int, v2 timestamptz default proctime());
```
Comment (BugenZhao) [inline e2e_test/batch/basic/table_with_default_columns.slt.part:47]: I'm not sure but IMO `now` is used with default column values (to follow the behavior of traditional databases) while `proctime` is for generated columns (to follow the behavior of streaming engines). So I suggest we may keep it untouched.

Though the support for `default now()` will be temporarily broken after #9762.

Comment (BugenZhao) [reply e2e_test/batch/basic/table_with_default_columns.slt.part:47]: We may utilize this to distinguish between `DEFAULT` and generated.

https://github.com/risingwavelabs/risingwave/blob/ca2109e65b24dd15edec9e6303ce8821b4a40f5b/src/frontend/src/binder/bind_context.rs#L73-L74

Comment (yuhao-su) [reply e2e_test/batch/basic/table_with_default_columns.slt.part]: done

File: e2e_test/ddl/table/generated_columns.slt.part [hunk 2]
```diff
@@ -113,3 +113,14 @@ drop materialized view mv;
 
 statement ok
 drop source t4;
+
+# create a table with generated column now
+statement error
```
Comment (BugenZhao) [inline e2e_test/ddl/table/generated_columns.slt.part:118]: Would you please fill the error message here?

File: src/frontend/src/binder/expr/function.rs [hunk 3]
```diff
@@ -922,6 +923,13 @@ impl Binder {
             ))
             .into());
         }
+        if matches!(self.context.clause, Some(Clause::GeneratedColumn)) {
```
Comment (BugenZhao) [inline src/frontend/src/binder/expr/function.rs:926]: Can we also use `clause` to get rid of `BindFor::Ddl`?

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM

File: src/frontend/src/binder/expr/function.rs (hunk 3 above)
Comment (yuhao-su) [reply src/frontend/src/binder/expr/function.rs:926]: Yes maybe in later prs

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**